airbyte-webapp/src/packages/cloud/views/settings/integrations/DbtCloudSettingsView.tsx

```diff
@@ -12,21 +12,39 @@ import styles from "./DbtCloudSettingsView.module.scss";
 
 export const DbtCloudSettingsView: React.FC = () => {
   const { mutate: submitDbtCloudIntegrationConfig, isLoading } = useSubmitDbtCloudIntegrationConfig();
+  const [hasValidationError, setHasValidationError] = useState<boolean>(false);
+  const [validationErrorMessage, setValidationErrorMessage] = useState<string | null>(null);
   return (
     <SettingsCard title={<FormattedMessage id="settings.integrationSettings.dbtCloudSettings" />}>
       <Content>
         <Formik
           initialValues={{
             serviceToken: "",
           }}
-          onSubmit={({ serviceToken }) => submitDbtCloudIntegrationConfig(serviceToken)}
+          onSubmit={({ serviceToken }) => {
+            setHasValidationError(false);
+            return submitDbtCloudIntegrationConfig(serviceToken, {
+              onError: (e) => {
+                setHasValidationError(true);
+
+                // I honestly don't know why I had to do this; for some reason, when I
+                // gave a `TError` type argument to the upstream `useMutation`, I start
+                // getting spurious "no matching overloads" errors.
+                // eslint-disable-next-line
+                setValidationErrorMessage((e as any)?.message?.replace("Internal Server Error: ", ""));
```

Comment:
Can we please also file a `technical-debt` issue for this place that the backend should still switch to a 400 (or likely better 422 since the request is technically correct just the token inside it is not a valid dbt token), so we don't forget that this still needs to be cleaned up in the backend?

Reply:
[Done](https://github.com/airbytehq/airbyte-cloud/issues/3553).